### PR TITLE
[O365]Moving edge processing to ingest pipelines

### LINKED ISSUE #666: Convert o365's edge processing to Ingest Node pipeline
# Convert edge processing to Ingest Node pipeline

This package uses Beats processors for some of its data processing. We want to
move that processing into the ingest node pipeline that is part of the package
to make reuse easier (e.g. data from Kafka could be routed through the pipeline).

### Data Streams
- audit (inputs: log, o365audit)
  - [ ] All beats processors are removed
  - [ ] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) added

### PR DESCRIPTION
**Pull request by P1llus**
## What does this PR do?

This PR removes all edge processing in favor of ingest pipelines

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.

## Related issues

- Closes #666

### CODE REVIEW

**Comment by P1llus** [comment]
run tests

**Comment by P1llus** [comment]
run tests

File: packages/o365/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -5,6 +5,998 @@ processors:
...
+      target_field: process.name
+      ignore_missing: true
+      if: ctx.event?.code == "ExchangeItem"
+  # AzureActiveDirectory Schema
+  - set: 
+      field: user.target.id
+      copy_from: o365audit.ObjectId
+      if: ctx.event?.code == "AzureActiveDirectory"
+  ## AzureActiveDirectory Schema new user
+  - set:
```
[the remark this answers is not in the document]
Comment (P1llus) [reply packages/o365/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:264]: Thanks for the feedback @leehinman. There is an ongoing discussion on exactly this, also for google_workspace, around things like performance and which way would be better.

If its okay with you I would like to do the same here, in which I leave it like this for now, as I feel it is better for performance, while keeping an eye on the overall discussion. If it turns out that there is no difference or the other way does not have any high impact on benchmarks I will go back and implement it on both.

**Review by marc-gr — APPROVED** [review]

**Pull request merged**